punx/github_handler.py

```diff
@@ -57,44 +58,38 @@
 # NXDL_CACHE_SUBDIR = GITHUB_NXDL_REPOSITORY + '-' + GITHUB_NXDL_BRANCH
 
 
-def get_GitHub_credentials(creds_file_name=""):
+def get_GitHub_credentials():
     """
-    get the Github Basic Authentication token from a file or environment
+    Get the Github API token from a file or environment.
 
     GitHub requests use an access token. The token is unique to a user and may
     be generated by visiting https://github.com/settings/tokens.
 
-    The token may be provided in the local file ``__github_creds__.txt`` or in
-    the environment variables GH_TOKEN or GITHUB_TOKEN. If found, the file is
-    parsed for ``username password token`` or ``token``. The token is assumed
-    to be the last whitepace-separated string in the file.
+    The token is provided in either of these environment variables: ``GH_TOKEN``
+    or ``GITHUB_TOKEN`` (searched in that order).
 
-    The search for the tokens checks in this order:
-        GH_TOKEN
-        GITHUB_TOKEN
-        ``creds_file_name``
-        ``[punx source directory]/__github_creds__.txt``
-        ``~/.config/__github_creds__.txt``
-        ``~/.config/punx/__github_creds__.txt``
-
-    The credentials file is not placed under version control since it has
-    GitHub credentials. Be sure to make the file readable only by the user and
-    not others.
-
-    Returns None if credentials are not found per above search.
+    Issues a warning and returns ``None`` if credentials are not found per above
+    search.
     """
+    # check for environment variables
     for variable in ["GH_TOKEN", "GITHUB_TOKEN"]:
-        if variable in os.environ:
-            return os.environ[variable]
-
-    for name in [
-        creds_file_name,
-        os.path.join(os.path.dirname(__file__), CREDS_FILE_NAME),
-        os.path.join(os.path.expanduser("~"), ".config", CREDS_FILE_NAME),
-        os.path.join(os.path.expanduser("~"), ".config", "punx", CREDS_FILE_NAME),
-    ]:
-        if os.path.exists(name):
-            return open(name, "r").read().strip().split()[-1]
+        token = os.environ.get(variable)
+        if token is not None:
+            return token
+
+    warnings.warn(
+        UserWarning(
+            "Did not find environment variables GH_TOKEN or GITHUB_TOKEN"
+        )
+    )
+    # TODO: move next conent to documentation.
+    # warnings.warn(
+    #     "Did not find environment variables GH_TOKEN or GITHUB_TOKEN"
+    #     " which provide the GitHub API token necessary to download"
+    #     " resources from GitHub through its API.  You may experience"
+    #     " restrictions on the amount of content that can be downloaded"
+    #     " over a chort intervale (such as an hour or so)."
+    # )
```

Comment:
I love a helpful warning/error message like this one. Why do you want to hide it in the docs?

Reply:
Thought it was too verbose.  After editing the `chort intervale` typo, I'll re-enable this one.